kratos/sources/model_part_io.cpp

```diff
@@ -262,12 +252,9 @@ namespace Kratos
             std::string element_name;
             auto elements_components = KratosComponents<Element>::GetComponents();
             
-            auto element_type = rThisElements.begin()->GetGeometry().GetGeometryType();
-            
             // Fisrt we do the first element
             for(auto it_comp = elements_components.begin(); it_comp != elements_components.end() ; it_comp++) {
-                if (IsSameElement<decltype(*(rThisElements.begin())), decltype(*(it_comp->second))>::value &&
-                    (element_type == (it_comp->second)->GetGeometry().GetGeometryType())) {
+                if(CompareElementsAndConditionsUtility::IsSame(*(rThisElements.begin()), *it_comp->second)) {
                     element_name = it_comp->first;
```

Comment:
If you have a utility for comparing elements I would add the geometry comparison to it.

Reply:
It already does